### PR TITLE
Add isEqual and hash for aggregate classes

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Firestore/Example/Tests/Integration/API/FIRCountTests.mm [hunk 1]
```diff
@@ -28,6 +28,38 @@ @interface FIRCountTests : FSTIntegrationTestCase
 
 @implementation FIRCountTests
 
+- (void)testEquals {
```
Comment (dconeybe) [inline Firestore/Example/Tests/Integration/API/FIRCountTests.mm:31]: nit: Name this method `testAggregateQueryEquals`

File: Firestore/Example/Tests/Integration/API/FIRCountTests.mm [hunk 2]
```diff
@@ -77,6 +109,41 @@ - (void)testCanRunCountWithOrderBys {
...
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"b"] count]];
+  FIRAggregateQuerySnapshot* snapshot1Same =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"b"] count]];
+
+  FIRAggregateQuerySnapshot* snapshot2 =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"a"] count]];
+  [self writeDocumentRef:[testCollection documentWithPath:@"d"] data:@{@"k" : @"a"}];
+  FIRAggregateQuerySnapshot* snapshot2Different =
+      [self readSnapshotForAggregate:[[testCollection queryWhereField:@"k" isEqualTo:@"a"] count]];
+
```
Comment (dconeybe) [inline Firestore/Example/Tests/Integration/API/FIRCountTests.mm:133]: Add a test case for two different `FIRAggregateQuerySnapshot` objects with the same result set but different queries. So far you've tested snapshots with the same AggregateQuery and same results, and the same AggregateQuery with different results. That third one is the only one missing to fill out the test coverage.

**Review by dconeybe — CHANGES_REQUESTED** [review]
Just a couple of minor comments. Otherwise, LGTM. Thanks for filling out this implementation!

File: Firestore/Example/Tests/Integration/API/FIRCountTests.mm (hunk 2 above)
Comment (wu-hui) [reply Firestore/Example/Tests/Integration/API/FIRCountTests.mm:133]: Done.

**Review by dconeybe — APPROVED** [review]
LGTM

**Pull request merged**